File: src/theme/tokens.js
// HALE — "Clean Dark" design tokens (v2) · SINGLE SOURCE OF TRUTH.
//
// Every token value lives here and ONLY here. Consumed by:
//   • tailwind.config.js  → className tokens (bg-bg, text-fg, bg-accent, …)
//   • src/theme/clean.ts  → typed runtime hexes (SVG / Skia / StatusBar / shadows)
//
// Source design: the approved Clean Dark bundle's styles.css (:root).
// Discipline (locked in the design chat): ONE emerald accent — one focal
// element + one primary CTA per screen; coral is SOS/danger ONLY; warm amber
// is buddy/"together"/referral ONLY.
//
// CommonJS so tailwind.config.js (Node) can require() it. Types in tokens.d.ts.

const cleanDark = {
  bg: '#0B0F0D', // app base — premium dark, never pure black
  bg2: '#0E1311',
  surface: '#151B18', // card
  surface2: '#1C2420', // elevated / input
  surface3: '#25302A', // pressed / hover
  stroke: 'rgba(255,255,255,0.07)', // hairline borders
  stroke2: 'rgba(255,255,255,0.13)',
  hi: 'rgba(255,255,255,0.04)', // top inner highlight on cards
  fg: '#EAF1EC', // primary text
  fg2: '#97A39B', // secondary
  fg3: '#616B64', // muted / labels
  fg4: '#424A45',
  accent: '#34D399', // THE focal emerald
  accent2: '#5EE3B0',
  accentDeep: '#1FA577',
  accentInk: '#052B1E', // text on accent
  accentSoft: 'rgba(52,211,153,0.12)',
  accentEdge: 'rgba(52,211,153,0.26)', // accent hairline ring
  accentGlow: 'rgba(52,211,153,0.35)',
  warm: '#F2B95C', // buddy / together / referral moments ONLY
  warmSoft: 'rgba(242,185,92,0.12)',
  warmEdge: 'rgba(242,185,92,0.28)',
  coral: '#FF6B5C', // SOS / relapse ONLY
  coralSoft: 'rgba(255,107,92,0.12)',
  coralEdge: 'rgba(255,107,92,0.30)',
  coralInk: '#2A0B07', // text on a coral button
  warmInk: '#2B1E07', // text on a warm button
  track: 'rgba(255,255,255,0.08)', // progress-bar trough
};

// Tailwind / NativeWind color object (className tokens), derived from cleanDark.
const tailwindColors = {
  bg: { DEFAULT: cleanDark.bg, 2: cleanDark.bg2 },
  surface: { DEFAULT: cleanDark.surface, 2: cleanDark.surface2, 3: cleanDark.surface3 },
  stroke: { DEFAULT: cleanDark.stroke, 2: cleanDark.stroke2 },
  hi: cleanDark.hi,
  fg: { DEFAULT: cleanDark.fg, 2: cleanDark.fg2, 3: cleanDark.fg3, 4: cleanDark.fg4 },
  accent: {
    DEFAULT: cleanDark.accent,
    2: cleanDark.accent2,
    deep: cleanDark.accentDeep,
    ink: cleanDark.accentInk,
    soft: cleanDark.accentSoft,
    edge: cleanDark.accentEdge,
    glow: cleanDark.accentGlow,
  },
  warm: { DEFAULT: cleanDark.warm, soft: cleanDark.warmSoft, edge: cleanDark.warmEdge, ink: cleanDark.warmInk },
  coral: { DEFAULT: cleanDark.coral, soft: cleanDark.coralSoft, edge: cleanDark.coralEdge, ink: cleanDark.coralInk },
  track: cleanDark.track,
};

// Clean Dark type ramp (from styles.css) — Sora everywhere.
const fontSize = {
  hero: ['88px', { lineHeight: '92px', letterSpacing: '-2.6px' }], // Sora clips below ~1.05lh
  h1: ['30px', { lineHeight: '36px', letterSpacing: '-0.75px' }],
  h2: ['23px', { lineHeight: '28px', letterSpacing: '-0.46px' }],
  h3: ['18px', { lineHeight: '23px', letterSpacing: '-0.18px' }],
  lead: ['16px', { lineHeight: '25px' }],
  body: ['15px', { lineHeight: '23px' }],
  eyebrow: ['12px', { lineHeight: '16px', letterSpacing: '1.56px' }], // 0.13em caps
};

// Rhythmic spacing — the 4pt design scale IS Tailwind's default scale; only the
// shared screen gutter is named.
const spacing = {
  gutter: '24px', // standard horizontal screen padding
};

const radius = {
  pill: '999px',
  panel: '22px', // r-card
  tile: '16px', // r-lg
  inset: '9px', // r-sm
  xl2: '26px', // r-xl (sheets)
  // 12px (r-md) == Tailwind's default `rounded-xl` — no custom key needed
};

module.exports = { cleanDark, tailwindColors, fontSize, spacing, radius };

File: apple/Hale/DesignSystem/Tokens.swift
import SwiftUI

// HALE — "Clean Dark" design tokens (v2). Ported verbatim from src/theme/tokens.js
// (the RN single source of truth). Dark-only. Discipline: ONE emerald accent per
// screen; coral = SOS/danger ONLY; warm amber = buddy/together/referral ONLY.
enum Tok {
    // MARK: Base / surfaces
    static let bg        = Color(hex: 0x0B0F0D)              // app base — premium dark, never pure black
    static let bg2       = Color(hex: 0x0E1311)
    static let surface   = Color(hex: 0x151B18)             // card
    static let surface2  = Color(hex: 0x1C2420)             // elevated / input
    static let surface3  = Color(hex: 0x25302A)             // pressed / hover
    static let stroke    = Color.rgba(255, 255, 255, 0.06)  // hairline borders (quiet)
    static let stroke2   = Color.rgba(255, 255, 255, 0.12)  // stronger divider / control edge
    static let hairline  = Color.rgba(255, 255, 255, 0.05)  // dividers between rows — barely there
    static let hi        = Color.rgba(255, 255, 255, 0.04)  // top inner highlight on cards

    // MARK: Text
    static let fg        = Color(hex: 0xEAF1EC)             // primary
    static let fg2       = Color(hex: 0x97A39B)             // secondary
    static let fg3       = Color(hex: 0x616B64)             // muted / labels
    static let fg4       = Color(hex: 0x424A45)

    // MARK: Accent (emerald) lane
    static let accent     = Color(hex: 0x34D399)            // THE focal emerald
    static let accent2    = Color(hex: 0x5EE3B0)
    static let accentDeep = Color(hex: 0x1FA577)
    static let accentInk  = Color(hex: 0x052B1E)            // text on accent
    static let accentSoft = Color.rgba(52, 211, 153, 0.12)
    static let accentEdge = Color.rgba(52, 211, 153, 0.26)  // accent hairline ring
    static let accentGlow = Color.rgba(52, 211, 153, 0.35)

    // MARK: Warm (buddy / together / referral) lane
    static let warm      = Color(hex: 0xF2B95C)
    static let warmSoft  = Color.rgba(242, 185, 92, 0.12)
    static let warmEdge  = Color.rgba(242, 185, 92, 0.28)
    static let warmInk   = Color(hex: 0x2B1E07)

    // MARK: Coral (SOS / relapse) lane
    static let coral     = Color(hex: 0xFF6B5C)
    static let coralSoft = Color.rgba(255, 107, 92, 0.12)
    static let coralEdge = Color.rgba(255, 107, 92, 0.30)
    static let coralInk  = Color(hex: 0x2A0B07)

    static let track     = Color.rgba(255, 255, 255, 0.08)  // progress-bar trough

    // MARK: Radii (pt)
    enum R {
        static let pill:  CGFloat = 999
        static let panel: CGFloat = 22   // r-card
        static let tile:  CGFloat = 16   // r-lg
        static let inset: CGFloat = 9    // r-sm
        static let xl2:   CGFloat = 26   // r-xl (sheets)
        static let md:    CGFloat = 12   // r-md (Tailwind default rounded-xl)
    }

    // MARK: Spacing & rhythm
    // One rhythm every screen inherits: a consistent gutter, larger breathing
    // room between sections, generous top padding, and a comfortable max content
    // width so nothing sprawls on big devices. (Restraint > density.)
    static let gutter:     CGFloat = 24   // standard horizontal screen padding
    static let section:    CGFloat = 28   // space between major sections in a screen
    static let sectionLg:  CGFloat = 32   // hero → first section (extra air)
    static let screenTop:  CGFloat = 20   // generous top padding under the safe area
    static let maxContent: CGFloat = 440  // comfortable reading width (clamps on iPad)
    static let cardPad:    CGFloat = 20   // default card inner padding
}
